accepted/2020/or-greater-defines/or-greater-defines.md

```diff
@@ -0,0 +1,218 @@
+`PlatformNotSupportedException`.
+
+Since she needs to provide custom implementations per OS, she wants to fail
+whenever a new TFM is added:
+
+```C#
+public static class Gps
+{
+    public static GpsCoordinate GetCoordinates()
+    {
+        #if ANDROID
+            return CallAndroidApi();
+        #elif IOS
+            return CallIOSApi();
+        #elif WINDOWS
+            return CallWindowsApi();
+        #elif NETSTANDARD
+            throw new PlatformNotSupportedException();
+        #else
+            #error Provide implementation
+        #endif
+    }
+}
+```
+
+### When running on a later version of .NET, use a newer API
+
+Zachariah is building a .NET imaging library. On .NET 5, we have added enough
+hardware intrinsics so that he can implement the blur functionality in a much
+faster way. However, he still wants to support .NET Standard 2.0, so he decides
+to use `#if` to multi-target for .NET 5.
+
+Since he checks for the minimum version of the framework that introduced the
+API, adding a new version wouldn't change this code, so he simply puts the
+fallback logic for the slow version in the `#else` branch, with no call to
+`#error`.
+
+```C#
+public class Picture
+{
+    public void Blur(float radius)
+    {
+        #if NET5_0_OR_HIGHER
+            FastBlurUsingHardwareIntrinsics();
+        #else
+            SlowBlurUsingSimpleMath();
+        #endif
+    }
+}
+```
+
+## Requirements
+
+### Goals
+
+* The new symbols don't change semantics of existing code
+* The new symbols allow expressing a `>=` relationship with version numbers
+* The new symbols apply to both frameworks as well as platforms
+* The new symbols are defined for all frameworks that are relevant for package
+  authors today, which includes:
+    - .NET 5
+    - .NET Core
+    - .NET Framework
+    - .NET Standard
+
+### Non-Goals
+
+* Making existing code more resilient to TFM changes
+
+## Design
+
+The SDK will define `XXX_OR_GREATER` variants for the following TFMs:
+
+* **.NET Framework**
+    - Applies to `net1.0`-`net4x` and the `NETFRAMEWORK`/`NET` symbols
+    - For example, `net4.8` will define `NETFRAMEWORK`, `NET48`,
+      `NET48_OR_GREATER` `NET472_OR_GREATER`, `NET471_OR_GREATER`, etc.
+* .NET Core
+    - Applies to `netcoreappX.Y` and the `NETCOREAPP` symbol
+    - For example, `netcoreapp3.1` will define `NETCOREAPP`, `NETCOREAPP3_1`,
+      `NETCOREAPP3_1_OR_GREATER`, `NETCOREAPP3_0_OR_GREATER` etc.
+* .NET 5 and later
+    - Applies to `netX.Y` and the `NET` symbol
+    - For example, `net6.0` will define `NET`, `NET6_0`, `NET6_0_OR_GREATER`,
+      `NET5_0_OR_GREATER` etc.
+    - This will also include the corresponding defines a .NET Core 3.1 successor
+      would have gotten. For example, `net5.0` will also define `NETCOREAPP`,
+      `NETCOREAPP3_1_OR_GREATER`, `NETCOREAPP3_0_OR_GREATER` etc. But it will
+      *neither* define a `NETCOREAPP5_0` or `NETCOREAPP5_0_OR_GREATER`.
```

Comment:
Does this go all the way back to `NETCOREAPP1_1_OR_GREATER`?

Reply:
> Does this go all the way back to NETCOREAPP1_1_OR_GREATER?

Yes, that's what the `etc.` was supposed to indicate. I've replaced that with an explicit range.